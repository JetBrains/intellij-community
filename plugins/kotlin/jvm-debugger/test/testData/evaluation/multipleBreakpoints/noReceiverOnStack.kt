package noReceiverOnStack

fun String.one() = 1

class A {
    var sum = 0

    fun addOne() {
        sum += 1
    }
}

fun buildSum(block: A.() -> Unit) = A().apply(block).sum

fun main(args: Array<String>) {
    // EXPRESSION: buildString { append("Hello!") }
    // RESULT: "Hello!": Ljava/lang/String;
    //Breakpoint!
    buildString { append("Hello!") }

    // EXPRESSION: buildSum { repeat(3) { addOne() } }
    // RESULT: 3: I
    //Breakpoint!
    buildSum { repeat(3) { addOne() } }

    // EXPRESSION: buildString { append(buildSum { buildString { addOne() } }) }
    // RESULT: "1": Ljava/lang/String;
    //Breakpoint!
    buildString { append(buildSum { buildString { addOne() } }) }

    // EXPRESSION: with("Hello!") { one() }
    // RESULT: 1: I
    //Breakpoint!
    with("Hello!") { one() }
}