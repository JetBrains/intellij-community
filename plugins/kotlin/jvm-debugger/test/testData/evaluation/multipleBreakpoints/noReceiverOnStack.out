LineBreakpoint created at noReceiverOnStack.kt:19
LineBreakpoint created at noReceiverOnStack.kt:24
LineBreakpoint created at noReceiverOnStack.kt:29
LineBreakpoint created at noReceiverOnStack.kt:34
Run Java
Connected to the target VM
noReceiverOnStack.kt:19
Compile bytecode for buildString { append("Hello!") }
noReceiverOnStack.kt:19
Compile bytecode for buildSum { repeat(3) { addOne() } }
noReceiverOnStack.kt:24
Compile bytecode for buildString { append(buildSum { buildString { addOne() } }) }
noReceiverOnStack.kt:24
Compile bytecode for with("Hello!") { one() }
noReceiverOnStack.kt:24
resuming noReceiverOnStack.kt:24
resuming noReceiverOnStack.kt:24
resuming noReceiverOnStack.kt:29
resuming noReceiverOnStack.kt:29
resuming noReceiverOnStack.kt:29
resuming noReceiverOnStack.kt:29
resuming noReceiverOnStack.kt:34
resuming noReceiverOnStack.kt:34
Disconnected from the target VM

Process finished with exit code 0
